python/cuml/cluster/hdbscan/prediction.pyx

```diff
@@ -255,8 +260,10 @@ def membership_vector(clusterer, points_to_predict, batch_size=4096, convert_dty
         in smaller batches of points in the prediction data. A batch size
         of 1000 computes distance based memberships for 1000 points at a
         time. The default batch_size is 4096. If the number of rows in
-        the prediction dataset is less than 4096, this defaults to the
-        number of rows.
+        the prediction data is less than 4096, this defaults to the
+        number of rows. If a batch size larger than the number of rows in
+        the prediction data is passed, the batch size used is the number
+        of rows in the prediction data.
```

Comment:
I would recommend the following wording of this doc-string here:

> Lowers memory requirement by computing distance-based membership
> in smaller batches of points in the prediction data. For example, a batch size
> of 1,000 computes distance based memberships for 1,000 points at a time. 
> The default batch size is 4,069.

I would argue that the fact that the batch size is reduced to the number of points to predict in case that it is smaller is self-evident and irrelevant to the user. If you want to keep that information in the docs, then I would recommend the following phrasing:

> The default batch size is 4,096 or the number of points to predict (whichever is smaller).